include/oneapi/dpl/pstl/hetero/algorithm_impl_hetero.h

```diff
@@ -705,12 +686,17 @@ __pattern_find_if(__hetero_tag<_BackendTag>, _ExecutionPolicy&& __exec, _Iterato
         return __last;
 
     using _Predicate = oneapi::dpl::unseq_backend::single_match_pred<_Pred>;
+    using _IndexType = std::make_unsigned_t<typename std::iterator_traits<_Iterator>::difference_type>;
+    using _TagType = __par_backend_hetero::__parallel_find_forward_tag<_IndexType>;
+    using __size_calc = oneapi::dpl::__par_backend_hetero::__min_size_calc;
```

Comment:
There is only one input range: let's use `__first_size_calc` to avoid calling `std::min` under the hood.